log.go

```diff
@@ -2,12 +2,119 @@ package didplc
 
 import (
 	"fmt"
+	"sync"
 	"time"
 
-	"github.com/bluesky-social/indigo/atproto/crypto"
 	"github.com/bluesky-social/indigo/atproto/syntax"
 )
 
+type OpStatus struct {
+	CreatedAt   time.Time // the only immutable field here
+	Nullified   bool
+	LastChild   string // CID
+	AllowedKeys []string
+}
+
+// Note: LogValidationContext is designed such that it could later be turned into an interface,
+// optionally backed by a db rather than in-memory
+// Note: ops are globally unique by CID, so opStatus map can be shared across all DIDs
+type LogValidationContext struct {
+	head     map[string]string    // DID -> CID, tracks most recent valid op for a particular DID
```

Comment:
because this type has required struct fields, I think it should have a "New" helper method (eg, `NewLogValidationContext()`) which handles initialization